### PR TITLE
Create 1.8 version and refactor build system to work across versions.

### PR DESCRIPTION
**Pull request by maxpowa**
Also a couple README changes to improve clarity, fix typos. 1.8 version loads fine (I had to make a minor change to ClientSync.java), CMM has been updated to 1.8 for a while now.

### CODE REVIEW

**Comment by superzanti** [comment]
Is a new build version necessary? This should work with any version of CMM as my mod is only based off the button ID. What exactly did you change?

I don't think having 2 separate  builds is needed. Though I could be wrong. Just explain yourself/reasoning a bit more.

**Comment by maxpowa** [comment]
1.8 moved cpw.mods.fml packages to net.minecraftforge.fml . If it weren't for that srgnames would resolve it all. I might be wrong too, I'd try a 1.7 build on 1.8 and see what happens first.

**Comment by superzanti** [comment]
Oh, you mean 1.8 minecraft

**Comment by maxpowa** [comment]
Yep, it builds both a 1.7.10 and a 1.8 version now.

**Comment by superzanti** [comment]
Awesome. If it works worth both then just update the main src folder and files and delete the 1.8 folder. If you do that I will pull.

**Comment by maxpowa** [comment]
Nonono, that's what I mean, the new setup builds 2 versions, a 1.7.10 using the 1.7.10 folder and a 1.8, using the 1.8 folder. If either folder is deleted, it won't build that version.

**Comment by superzanti** [comment]
So what is the fundamental line of code that changes that makes it work with 1.8 but not with 1.7.10? I don't have the time to diff it to check.

File: 1.8/src/main/java/com/superzanti/serversync/ServerSyncRegistry.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.config.Configuration;
+import net.minecraftforge.common.config.Property;
+
+import org.apache.logging.log4j.Logger;
+
+import net.minecraftforge.fml.common.Mod;
```
Comment (maxpowa) [inline 1.8/src/main/java/com/superzanti/serversync/ServerSyncRegistry.java]: All of these imports in the main mod file for example. These are cpw.mods.fml in 1.7.10 and net.minecraftforge.fml in 1.8. This is just one example, there's many other files that you're using FML things in.

**Comment by superzanti** [comment]
Could we use a compiler define instead? That way multiple sources are more required?

**Comment by rheimus** [comment]
I'm considering making Serversync standalone on the server side as well, this would bypass the problem altogether but it could make the application less approachable/plug'n'play.

This combined with the idea of allowing the user to specify directories to include in the sync process would open up the possibility of making Serversync a multi-purpose universal sync tool.

Alternatively I could just look into making multiple builds and update the relatively small part of the application still using forge code.

**Comment by rheimus** [comment]
In the meantime, the MultiVersion branch supports updating to any version desired.

**Pull request closed without merging**